### ISSUE TITLE
test_snapshots and test_show fail

### ISSUE THREAD

**Issue by mcepl** [mpld3/mpld3#505]
When packaging mpld3 for openSUSE, we run test suite to prove at least rudimentary functionality of our package and it fails on two test cases:
```
[    7s] =================================== FAILURES ===================================
[    7s] ________________________________ test_snapshots ________________________________
[    7s]
[    7s]     def test_snapshots():
[    7s]         print("Checking test_plots against stored snapshots")
[    7s]         plot_files = glob.glob(TEST_PLOT_FILES)
[    7s]
[    7s]         expected_snapshots = {}
[    7s]         for plot_file in plot_files:
[    7s]             plot_snapshot = export.snapshot_path(plot_file, TEST_PLOT_SNAPSHOT_DIR)
[    7s]             if not os.path.isfile(plot_snapshot):
[    7s]                 continue
[    7s]             expected_snapshots[plot_snapshot] = plot_file
[    7s]
[    7s] >       got = export.snapshot_mpld3_plots_consecutive(expected_snapshots.values())
[    7s] E       TypeError: snapshot_mpld3_plots_consecutive() missing 1 required positional argument: 'output_folder'
[    7s]
[    7s] mpld3/tests/test_d3_snapshots.py:25: TypeError
[    7s] ----------------------------- Captured stdout call -----------------------------
[    7s] Checking test_plots against stored snapshots
[    7s] __________________________________ test_show ___________________________________
[    7s]
[    7s]     def test_show():
[    7s]         """Test mpld3.show() using a mock server"""
[    7s]
[    7s]         class MockRequest(object):
[    7s]             def makefile(self, *args, **kwargs):
[    7s]                 return IO(b"GET /")
[    7s]
[    7s]         class MockServer(object):
[    7s]             def __init__(self, ip_port, Handler):
[    7s]                 handler = Handler(MockRequest(), ip_port[0], self)
[    7s]
[    7s]             def serve_forever(self):
[    7s]                 pass
[    7s]
[    7s]             def server_close(self):
[    7s]                 pass
[    7s]
[    7s]         # Any exceptions in the show() routine will pop up here
[    7s]         plt.plot([1, 2, 3])
[    7s] >       show(open_browser=False, http_server=MockServer)
[    7s]
[    7s] mpld3/tests/test_show.py:36:
[    7s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[    7s] mpld3/_display.py:370: in show
[    7s]     serve(html, ip=ip, port=port, n_retries=n_retries, files=files,
[    7s] mpld3/_server.py:103: in serve
[    7s]     srvr = http_server((ip, port), Handler)
[    7s] mpld3/tests/test_show.py:26: in __init__
[    7s]     handler = Handler(MockRequest(), ip_port[0], self)
[    7s] /usr/lib64/python3.9/socketserver.py:747: in __init__
[    7s]     self.handle()
[    7s] /usr/lib64/python3.9/http/server.py:427: in handle
[    7s]     self.handle_one_request()
[    7s] /usr/lib64/python3.9/http/server.py:415: in handle_one_request
[    7s]     method()
[    7s] mpld3/_server.py:38: in do_GET
[    7s]     self.wfile.write("<html><head>"
[    7s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[    7s]
[    7s] self = <socketserver._SocketWriter object at 0x7f4544f8f700>
[    7s] b = b'<html><head><title>mpld3 plot</title></head><body>\n'
[    7s]
[    7s]     def write(self, b):
[    7s] >       self._sock.sendall(b)
[    7s] E       AttributeError: 'MockRequest' object has no attribute 'sendall'
[    7s]
[    7s] /usr/lib64/python3.9/socketserver.py:826: AttributeError
[    7s] ----------------------------- Captured stderr call -----------------------------
[    7s] 1 - - [07/Feb/2022 02:07:35] "GET /" 200 -
[    7s] =============================== warnings summary ===============================
[    7s] mpld3/utils.py:24
[    7s]   /home/abuild/rpmbuild/BUILD/mpld3-0.5.7/mpld3/utils.py:24: DeprecationWarning: invalid escape sequence \s
[    7s]     return not re.search('\s', objid)
[    7s]
[    7s] mpld3/utils.py:26
[    7s]   /home/abuild/rpmbuild/BUILD/mpld3-0.5.7/mpld3/utils.py:26: DeprecationWarning: invalid escape sequence \-
[    7s]     return bool(re.match("^[a-zA-Z][a-zA-Z0-9\-\.\:\_]*$", objid))
[    7s]
[    7s] mpld3/_display.py:246
[    7s]   /home/abuild/rpmbuild/BUILD/mpld3-0.5.7/mpld3/_display.py:246: DeprecationWarning: invalid escape sequence \s
[    7s]     elif re.search('\s', figid):
[    7s]
[    7s] mpld3/tests/test_elements.py::test_markers
[    7s]   /home/abuild/rpmbuild/BUILD/mpld3-0.5.7/mpld3/tests/test_elements.py:32: UserWarning: color is redundantly defined by the 'color' keyword argument and the fmt string "^k" (-> color='k'). The keyword argument will take precedence.
[    7s]     ax.plot(np.arange(10), np.random.random(10),
[    7s]
[    7s] -- Docs: https://docs.pytest.org/en/stable/warnings.html
[    7s] =========================== short test summary info ============================
[    7s] FAILED mpld3/tests/test_d3_snapshots.py::test_snapshots - TypeError: snapshot...
[    7s] FAILED mpld3/tests/test_show.py::test_show - AttributeError: 'MockRequest' ob...
[    7s] =================== 2 failed, 14 passed, 4 warnings in 1.68s ===================
```

Complete [build log.txt](https://github.com/mpld3/mpld3/files/8011834/_log.txt) with all versions of packages used and steps taken.

Also, while looking at the package, I found that it still uses (completely unnecessarily) nose for importing SkipTest exception (which is already in the standard library), so I have create this patch to remove the dependency on already dead project nose:
```diff
---
 mpld3/mplexporter/tests/test_basic.py    |    2 +-
 mpld3/test_plots/test_pandas_timeaxis.py |    2 +-
 mpld3/tests/test_d3_snapshots.py         |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

--- a/mpld3/mplexporter/tests/test_basic.py
+++ b/mpld3/mplexporter/tests/test_basic.py
@@ -1,7 +1,7 @@
 import matplotlib
 import numpy as np
 from distutils.version import LooseVersion
-from nose.plugins.skip import SkipTest
+from unittest import SkipTest
 from numpy.testing import assert_warns

 from ..exporter import Exporter
--- a/mpld3/test_plots/test_pandas_timeaxis.py
+++ b/mpld3/test_plots/test_pandas_timeaxis.py
@@ -7,7 +7,7 @@ def create_plot():
     try:
         import pandas as pd
     except Exception:
-        from nose import SkipTest
+        from unittest import SkipTest
         raise SkipTest("pandas not installed")
     df2_index = pd.date_range(start="2010-01-01", periods=100, freq='D')
     df2 = pd.DataFrame({'a': range(100)}, index=df2_index)
--- a/mpld3/tests/test_d3_snapshots.py
+++ b/mpld3/tests/test_d3_snapshots.py
@@ -4,7 +4,7 @@ import glob
 import matplotlib
 from . import export

-from nose.plugins.skip import SkipTest
+from unittest import SkipTest
 matplotlib.use('Agg')

 TEST_PLOT_FILES  = os.path.join(mpld3.BASE_PATH, 'mpld3/test_plots/*.py')
```

**Comment by vladh** [comment]
Thank you @mcepl for the issue and the patch. This is on my to-do list, and I'll apply and test your patch, as well as fixing whatever problem is present in those tests, ASAP.

**Comment by vladh** [comment]
Fixed in fa3aeb, will be released in 0.5.8. Took me a while. :)

**Comment by mcepl** [comment]
Works for me with 0.5.8. Thank you!
